core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,198 @@
+    async fn dispatch(&self) -> anyhow::Result<()> {
+        let mut conn = self.pool.connection_tagged("da_dispatcher").await?;
+        let batches = conn
+            .data_availability_dal()
+            .get_ready_for_da_dispatch_l1_batches(self.config.max_rows_to_dispatch() as usize)
+            .await?;
+        drop(conn);
+
+        for batch in batches {
+            let dispatch_latency = METRICS.blob_dispatch_latency.start();
+            let dispatch_response = retry(self.config.max_retries(), batch.l1_batch_number, || {
+                self.client
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to dispatch a blob with batch_number: {}, pubdata_len: {}",
+                    batch.l1_batch_number,
+                    batch.pubdata.len()
+                )
+            })?;
+            let dispatch_latency_duration = dispatch_latency.observe();
+
+            let sent_at =
+                NaiveDateTime::from_timestamp_millis(Utc::now().timestamp_millis()).unwrap();
+
+            let mut conn = self.pool.connection_tagged("da_dispatcher").await?;
+            conn.data_availability_dal()
+                .insert_l1_batch_da(
+                    batch.l1_batch_number,
+                    dispatch_response.blob_id.as_str(),
+                    sent_at,
+                )
+                .await?;
+            drop(conn);
+
+            METRICS
+                .last_dispatched_l1_batch
+                .set(batch.l1_batch_number.0 as usize);
+            METRICS.blob_size.observe(batch.pubdata.len());
+            tracing::info!(
+                "Dispatched a DA for batch_number: {}, pubdata_size: {}, dispatch_latency: {dispatch_latency_duration:?}",
+                batch.l1_batch_number,
+                batch.pubdata.len(),
+            );
+        }
+
+        Ok(())
+    }
+
+    /// Polls the data availability layer for inclusion data, and saves it in the database.
+    async fn poll_for_inclusion(&self) -> anyhow::Result<()> {
+        let mut conn = self.pool.connection_tagged("da_dispatcher").await?;
+        let blob_info = conn
+            .data_availability_dal()
+            .get_first_da_blob_awaiting_inclusion()
+            .await?;
+        drop(conn);
+
+        if let Some(blob_info) = blob_info {
+            let inclusion_data = self
+                .client
+                .get_inclusion_data(blob_info.blob_id.as_str())
+                .await
+                .with_context(|| {
+                    format!(
+                        "failed to get inclusion data for blob_id: {}, batch_number: {}",
+                        blob_info.blob_id, blob_info.l1_batch_number
+                    )
+                })?;
+
+            let mut conn = self.pool.connection_tagged("da_dispatcher").await?;
```

Comment:
Nit: Doesn't matter much, but it looks more idiomatic to acquire a connection *after* the check that `inclusion_data` is `Some(_)`.